dogstatsd.mli

```diff
@@ -0,0 +1,140 @@
+module Base : sig
+  module Tag : sig
+    type t = string * string (** (tag-name, value) *)
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#metrics *)
+  module Metric : sig
+    module Counter : sig
+      type t =
+        [ `Increment
+        | `Decrement
+        | `Value of int ]
+    end
+
+    type typ =
+      [ `Counter of Counter.t
+      | `Gauge of float
+      | `Timer of float
+      | `Histogram of int
+      | `Set of string ]
+
+    type t =
+      { metric_name : string
+      ; metric : typ
+      ; sample_rate : float option
```

Comment:
The documentation says:
> Sample rates only work with counter, histogram, and timer metrics

Should we encode the sample rate into the typ perhaps? Too heavy handed? Just make a note in the interface and leave the interface it as is?

Reply:
>Just make a note in the interface and leave the interface it as is?
Going to do this.